board.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Orientation is a type that indicates in which direction the piece is moved.
+type Orientation int
+
+// Location is a type that represents a 2 dimensional coordinate.
+type Location struct {
+	x int
+	y int
+}
+
+// Direction constants.
+const (
+	Up Orientation = iota
+	Right
```

Comment:
Yeah, it's one of those weird Golang things.  In the absence of an assignment, they are assigned the value of the above element + 1.